test/integration/cdi/example-driver/app/kubeletplugin.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	cdiapi "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	specs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"k8s.io/klog/v2"
+	drapbv1 "k8s.io/kubernetes/pkg/kubelet/apis/dra/v1alpha1"
+	plugin "k8s.io/kubernetes/test/integration/cdi/example-driver/kubeletplugin"
+)
+
+type pluginConfig struct {
+	cdiDir                 string
+	driverName             string
+	draAddress             string
+	pluginRegistrationPath string
+}
+
+type examplePlugin struct {
+	config             pluginConfig
+	resourceHandleKeys []string
+}
+
+var resourceHandleKeys = []string{"some", "another"}
+
+// newExamplePlugin returns an initialized examplePlugin instance.
+func newExampleDriver(config pluginConfig) (*examplePlugin, error) {
+	examplePlugin := &examplePlugin{
+		config: config,
+	}
+	examplePlugin.resourceHandleKeys = makePossibleResourceHandleKeys(resourceHandleKeys)
+	return examplePlugin, nil
+}
+
+// makePossibleResourceHandleKeys returns keys prefixed by "user_" and "admin_".
+func makePossibleResourceHandleKeys(keys []string) []string {
```

Comment:
@hj-johannes-lee Is this restriction and validation code for resource handle really needed in this example? I'd remove it to make the code more understandable. After all we just need to set env vars out of all keys and values from the handle, right?

Reply:
I do not know then,,, what @pohly meant about 'validation'...

Reply:
Can you ask him or at least point me out to his comment about it? Anyway, it's just my opinion. We don't need to overload simple example code with anything that complicates it. @pohly opinion has higher priority as he's the owner of the target branch.

Reply:
Unmarshal into map[string]string, return the error if that fails. That's enough validation. Your original code manually parsed the ResourceHandle without returning an error.

Reply:
@hj-johannes-lee as we set keys of unmarshalled map as a env variable names we may need to validate if they follow env variable naming convention. However, that validation can be already done on the CDI side. Please, check.

Reply:
Ah,,, if what @pohly meant is just the validation inside the 'unmarshal', then... I misunderstood...

Since I knew that if the 'invalid' form of input is made, then even before plugin does something, already it is not possible to do "kubectl apply -f pod.yaml"